libnopegl/src/pipeline_compat.c

```diff
@@ -122,6 +121,51 @@ static int init_blocks_buffers(struct pipeline_compat *s, const struct pipeline_
     return 0;
 }
 
+static void free_bindgroup(void *user_arg, void *data)
+{
+    struct bindgroup *bindgroup = data;
+    ngli_bindgroup_freep(&bindgroup);
+}
+
```

Comment:
I think this is incorrect; I believe you need this:
```diff
diff --git a/libnopegl/src/pipeline_compat.c b/libnopegl/src/pipeline_compat.c
index 43f261609..80692787f 100644
--- a/libnopegl/src/pipeline_compat.c
+++ b/libnopegl/src/pipeline_compat.c
@@ -123,8 +123,8 @@ static int init_blocks_buffers(struct pipeline_compat *s, const struct pipeline_

 static void free_bindgroup(void *user_arg, void *data)
 {
-    struct bindgroup *bindgroup = data;
-    ngli_bindgroup_freep(&bindgroup);
+    struct bindgroup **bindgroupp = data;
+    ngli_bindgroup_freep(bindgroupp);
 }

 static int grow_bindgroup_array(struct pipeline_compat *s)
```

Reply:
Thanks for pointing that out. Fixed.